app/Controllers/feedController.php

```diff
@@ -734,19 +734,14 @@ public function reloadAction() {
 
 		//Extract all feed entries from database, load complete content and store them back in database.
 		$entries = $entryDAO->listWhere('f', $feed_id, FreshRSS_Entry::STATE_ALL, 'DESC', 0);
-
-		//We need another DB connection in parallel
-		Minz_ModelPdo::$usesSharedPdo = false;
-		$entryDAO2 = FreshRSS_Factory::createEntryDao();
+		$entries = iterator_to_array($entries);	//TODO: Improve
 
 		foreach ($entries as $entry) {
 			if ($entry->loadCompleteContent(true)) {
-				$entryDAO2->updateEntry($entry->toArray());
+				$entryDAO->updateEntry($entry->toArray());
```

Comment:
I've used only that modification and it works. There might be something fishy about the locking mechanism.

Reply:
This line enough does not seem enough for SQLite when dealing with a larger number of entries

Reply:
Update: scratch that - seems to work after all :-)